test/e2e/azure.go

```diff
@@ -176,3 +270,10 @@ func getAccessToken(cfg *clusterConfig) (*adal.ServicePrincipalToken, error) {
 	}
 	return nil, fmt.Errorf("no credentials provided for Azure API")
 }
+
+func getPrivateZoneFlag() bool {
```

Comment:
So are we going to run the Azure test twice, once with this environment variable set and once without?

Reply:
yes, we need to test for Azure ( public and private ) zones, each of them has different client package and setup .. so we use this flag to run the test one time for the public and another one for private